components/admin_console/billing/billing_subscriptions.tsx

```diff
@@ -51,12 +52,27 @@ const BillingSubscriptions: React.FC<Props> = () => {
     const currentUser = useSelector((state: GlobalState) => getCurrentUser(state));
     const isCloud = useSelector((state: GlobalState) => getLicense(state).Cloud === 'true');
     const subscription = useSelector((state: GlobalState) => state.entities.cloud.subscription);
+    const customer = useSelector((state: GlobalState) => state.entities.cloud.customer);
     const products = useSelector((state: GlobalState) => state.entities.cloud.products);
     const getCategory = makeGetCategory();
     const preferences = useSelector<GlobalState, PreferenceType[]>((state) => getCategory(state, Preferences.ADMIN_CLOUD_UPGRADE_PANEL));
 
     const contactSalesLink = useSelector((state: GlobalState) => getCloudContactUsLink(state, InquiryType.Sales));
 
+    const isCreditCardExpired = () => {
+        if (!customer) {
+            return false;
+        }
+
+        // Will developers ever learn? :D
+        const expiryYear = customer.payment_method.exp_year + 2000;
+        const lastExpiryDate = new Date(expiryYear, customer.payment_method.exp_month - 1, 1);
+        lastExpiryDate.setMonth(lastExpiryDate.getMonth() + 1);
```

Comment:
Why the -1 for the month and then add it back?

Reply:
when you define a `Date` object, the month value starts at 0 for some reason, so instead of 1 being January, 0 is January. But we store January as 1, so that's why we need the -1.

But then we have to increment it to get the actual expiry date of the card, since the card is usually good until the end of the month. So if your card expiry day is October 2021, you should be able to use it until October 31st, 2021.

I know it's redundant, but it's clear in terms of what operations are happening. I can remove the redundancy if you'd like and just add a comment stating the above.

Reply:
Comment would be great. Thanks!

Reply:
Done! Also moved out the expired function to a utils file.